public/scripts/utils.js

```diff
@@ -33,6 +34,74 @@ export function isValidUrl(value) {
     }
 }
 
+/**
+ * Converts string to a value of a given type. Includes pythonista-friendly aliases.
+ * @param {string|SlashCommandClosure} value String value
+ * @param {string} type Type to convert to
+ * @returns {any} Converted value
+ */
+export function convertValueType(value, type) {
+    if (value instanceof SlashCommandClosure || typeof type !== 'string') {
+        return value;
+    }
+
+    type = type.trim().toLowerCase();
+
+    switch (type) {
+        case 'string':
+        case 'str':
+            return String(value);
```

Comment:
Still not sure if we should keep "string" here. If it's the default, why do you provide an option to change the type to string?
I mean, it doesn't hurt. But especially with the upcoming debate on not stripping quotes around unnamed arguments, this will lead to inconsistent and unexpected behavior in my eyes, that is not really needed to add.
Already explained on the discord, but here again:

```
/setvar key=obj index=num type=number 1337
/setvar key=obj index=heh type=booelan true
/setvar key=obj index=arr type=array ["Test", "Blah", "Oke"]
/setvar key=obj index=obj type=object {"myVal": "Hello World!"}
/setvar key=obj index=str type=string "Hello World!"
```

From a programmers perspective my expectation would be that the surrounding quotes were meant to treat this as a string. All other types also accept valid JSON syntax. So using `"Hello World!"` should fill the object property with the string `Hello World!`.